optaplanner-quarkus-integration/optaplanner-quarkus/deployment/src/main/java/org/optaplanner/quarkus/deployment/GizmoMemberAccessorEntityEnhancer.java

```diff
@@ -117,6 +120,9 @@ public static String generateFieldAccessor(AnnotationInstance annotationInstance
                 .classOutput(classOutput)
                 .build();
 
+        classCreator.addAnnotation(Singleton.class);
```

Comment:
Is there any reason for not using the typical `@ApplicationScoped`?

One thing worries me more, though: we are exposing the generated GizmoMemberAccessor as a public bean. Any user can just `@Inject` it into his/her code.

Reply:
No; I can change it to `@ApplicationScoped`

In regards to the "Any user can just `@Inject`", I think that a non-issue: they could already do that via reflection. Additionally, the member accessors are stateless: meaning they can safely be reused anywhere. (They will also probably run into the issue where there are multiple MemberAccessors, and need to lookup the Gizmo generated classname)